cmd/kube-apiserver/app/server.go

```diff
@@ -254,14 +257,8 @@ func CreateNodeDialer(s completedServerRunOptions) (tunneler.Tunneler, *http.Tra
 		nodeTunneler = tunneler.New(s.SSHUser, s.SSHKeyfile, healthCheckPath, installSSHKey)
 
 		// Use the nodeTunneler's dialer when proxying to pods, services, and nodes
-		proxyDialerFn = nodeTunneler.Dial
+		proxyTransport.DialContext = nodeTunneler.Dial
 	}
-	// Proxying to pods and services is IP-based... don't expect to be able to verify the hostname
-	proxyTLSClientConfig := &tls.Config{InsecureSkipVerify: true}
-	proxyTransport := utilnet.SetTransportDefaults(&http.Transport{
```

Comment:
this is not equivalent, at least with respect to the CIDR-capable proxy:

https://github.com/kubernetes/kubernetes/blob/4a1da48d116b51089e1040016447a74858841037/staging/src/k8s.io/apimachinery/pkg/util/net/http.go#L92-L96

Reply:
actually, I take that back... go1.11 added some level of support for that:

https://golang.org/doc/go1.11#net/http:

> ProxyFromEnvironment, which is used by the DefaultTransport, now supports CIDR notation and ports in the NO_PROXY environment variable.

we do need to verify the implementation is compatible with what we had previously

Reply:
Golang SDK (v1.12): https://github.com/golang/net/blob/release-branch.go1.12/http/httpproxy/proxy.go#L218
`c.NoProxy` being set via the env `NO_PROXY` or `no_proxy`

apimachinery: https://github.com/kubernetes/apimachinery/blob/master/pkg/util/net/http.go#L275

From a first look it looks like the go SDK implementation covers what `NewProxierWithNoProxyCIDR` is doing + more (domains & ports)